packages/api/src/hooks/p2p/useCreateAdvertiser.ts

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useMemo } from 'react';
+import useAuthorize from '../useAuthorize';
+import useMutation from '../../useMutation';
+import useInvalidateQuery from '../../useInvalidateQuery';
+
+type TAdvertiseCreatePayload = {
+    name: string;
+};
+
+const useCreateAdvertiser = () => {
+    const invalidate = useInvalidateQuery();
+    const { isSuccess: isUserAuthorized } = useAuthorize();
+    const { data, mutate, ...rest } = useMutation('p2p_advertiser_create', {
+        onSuccess: () => {
+            invalidate('p2p_advertiser_list');
+            invalidate('p2p_advertiser_info');
+        },
+    });
+
+    const createAdvertiser = useCallback(
+        (payload: TAdvertiseCreatePayload) => {
+            if (isUserAuthorized) mutate({ payload });
+        },
+        [isUserAuthorized, mutate]
+    );
```

Comment:
I dont think we should be handling if user authorize or not, if we only call the mutation when we are authorized 🤔

Reply:
you mean like this?
```suggestion
    const createAdvertiser = useCallback(
        (payload: TAdvertiseCreatePayload) => {
            mutate({ payload });
        },
        [mutate]
    );
```

Reply:
Yes correct @shayan-deriv 🙇

Reply:
done ✅